slang-checks/src/main/java/org/sonarsource/slang/checks/BadFunctionNameCheck.java

```diff
@@ -19,28 +19,42 @@
  */
 package org.sonarsource.slang.checks;
 
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Pattern;
+import org.sonar.check.Rule;
+import org.sonar.check.RuleProperty;
 import org.sonarsource.slang.api.FunctionDeclarationTree;
 import org.sonarsource.slang.api.IdentifierTree;
 import org.sonarsource.slang.checks.api.InitContext;
 import org.sonarsource.slang.checks.api.SlangCheck;
-import java.util.regex.Pattern;
-import org.sonar.check.Rule;
-import org.sonar.check.RuleProperty;
 
 @Rule(key = "S100")
 public class BadFunctionNameCheck implements SlangCheck {
 
-  private static final String DEFAULT_FORMAT = "^[a-z][a-zA-Z0-9]*$";
+  public static final String DEFAULT_FORMAT = "^[a-z][a-zA-Z0-9]*$";
+
+  private static final Map<String, String> DEFAULT_BY_LANGUAGE;
+  static {
+    HashMap<String, String> defaults = new HashMap<>();
```

Comment:
wdyt to have `enum` for languages in the `checks` module and use `EnumMap` here?

Reply:
I'll leave this for now as is, and we'll improve that in another PR